File: plateforme/offer-pro-service/src/main/java/tn/esprit/Dtos/CompanyDto.java
package tn.esprit.Dtos;


import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.Entitys.ActivityArea;
import tn.esprit.Entitys.StateRegion;

import javax.persistence.Column;
import java.util.ArrayList;
import java.util.List;



@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
//id  name description state  city zipCode  address phone email logo activity_area
public class CompanyDto  {
    long id;
    String username;
    String name;
    String description;
    StateRegion state;
    String city;
    int zipCode;
    String address;
    int phone;
    String email;
    String linkedIn;
    String website;
    ActivityArea activity;
    AttachmentDto logo;
    AttachmentDto cover;
    List<RecruitmentOfferDto> recruitmentOffersDto = new ArrayList<RecruitmentOfferDto>();
}

File: webOfferPro/src/app/models/offer_pro_model/CompanyDto.ts
import { StateRegion } from "../user_model/StateRegion";
import { ActivityArea } from "./ActivityArea";
import { AttachementDto } from "./AttachementDto";  
import { RecruitmentOfferDto } from "./RecruitmentOfferDto";

export class CompanyDto {
    id! : number ;
    name! : string ;
    username! : string ;
    description! : string ;
    state! : StateRegion;  
    city! : string ;
    zipCode! : number;  
    address! : string ;
    phone!: number;   
    email! : string ;
    linkedIn! : string ;
    website! : string ;
    activity! : ActivityArea;
    logo : AttachementDto = new AttachementDto();
    cover : AttachementDto = new AttachementDto();
    recruitmentOffersDto : RecruitmentOfferDto[] = new Array<RecruitmentOfferDto>();  
}